app/upload.js

```diff
@@ -1,118 +1,159 @@
-define(['app/database', 'jquery', 'jszip', 'bootstrap'], function (db, $, jszip, bootstrap) {
-	window.JSZip = jszip;
-  requirejs(['xlsx'], function (XLSX) {
-		var XW = {
-			/* worker message */
-			msg : 'xlsx',
-			/* worker scripts */
-			rABS : './node_modules/xlsx/xlsxworker2.js',
-		};
-
-		var rABS = typeof FileReader !== "undefined" && typeof FileReader.prototype !== "undefined" && typeof FileReader.prototype.readAsBinaryString !== "undefined";
-
-		function ab2str(data) {
-			var o = "",
-			l = 0,
-			w = 10240;
-			for (; l < data.byteLength / w; ++l)
-				o += String.fromCharCode.apply(null, new Uint16Array(data.slice(l * w, l * w + w)));
-			o += String.fromCharCode.apply(null, new Uint16Array(data.slice(l * w)));
-			return o;
-		}
-
-		function s2ab(s) {
-			var b = new ArrayBuffer(s.length * 2),
-			v = new Uint16Array(b);
-			for (var i = 0; i != s.length; ++i) {
-				v[i] = s.charCodeAt(i);
-			}
-			return [v, b];
-		}
-
-		function xw_xfer(data, cb) {
-			var worker = new Worker(XW.rABS);
-			worker.onmessage = function (e) {
-				switch (e.data.t) {
-				case 'ready':
-					break;
-				case 'e':
-					console.error(e.data.d);
-					break;
-				default:
-					xx = ab2str(e.data).replace(/\n/g, "\\n").replace(/\r/g, "\\r");
-					console.log("done");
-					cb(JSON.parse(xx));
-					break;
+define(['app/database', 'jquery', 'bootstrap', 'xlsx', 'parsley', 'bootstrap-select'], function (db, $, bootstrap, XLSX, parsley, select) {
+	var X = XLSX;
+	var saveData;
+	var correctUpload = false;
+
+	// Script for evaluating the input of the upload form
+	$(function () {
+		$(".selectpicker").selectpicker();
+		window.Parsley.on('field:error', function() {
+				if (this.$element.is("select")) {
+						this.$element.parent().children('.selectpicker').selectpicker('setStyle', 'alert-danger').selectpicker('refresh');
 				}
-			};
-			var val = s2ab(data);
```

Comment:
What is that this function s2ab does? If you wrote it yourselves, shame on you and fix it, if it is a library shame on them but keep on using it like this ;-).